### ISSUE TITLE
 This application failed to start because it could not find or load the Qt platform plugin "xcb" in "".

### ISSUE THREAD

**Issue by rm-minus-r-star** [OpenShot/openshot-qt#1809]
Installed from ppa on Ubuntu 18.04 on Lenovo Thinkpad P50, with both the stable and daily, and launching via cli:                                                                

> Loaded modules from installed directory: /usr/lib/python3/dist-packages/openshot_qt
>       launch:INFO ------------------------------------------------
>       launch:INFO    OpenShot (version 2.4.2)
>       launch:INFO ------------------------------------------------
> This application failed to start because it could not find or load the Qt platform plugin "xcb"
> in "".
> 
> Available platform plugins are: eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.
> 
> Reinstalling the application may fix this problem.
> Aborted (core dumped)

**Comment by peanutbutterandcrackers** [comment]
Like it says, could you please reinstall the application?

Could you please also [attach the logs](https://github.com/OpenShot/openshot-qt/pull/1726#issuecomment-403209077), thanks!

(@ferdnyc should be able to help out, though; he helped me with a qt issue some time ago)

**Comment by rm-minus-r-star** [comment]
Yes, I tried numerous re-installations, the last of which was to switch to the daily source as mentioned.

I tried further since that post to `sudo apt install qt5*, and the re-install openshot-qt again, but again with no joy:

> user-name@P50:~$ sudo apt-get install openshot-qt | tee openshot-qt_install-log.txt
> [sudo] password for user-name: 
> Reading package lists...
> Building dependency tree...
> Reading state information...
> Suggested packages:
>   openshot-qt-doc
> The following packages will be upgraded:
>   openshot-qt
> 1 to upgrade, 0 to newly install, 0 to remove and 116 not to upgrade.
>                                                                      Need to get 0 B/53.1 MB of archives.
>                                                                                                          After this operation, 5,120 B of additional disk space will be used.
> (Reading database ... 200846 files and directories currently installed.)g database ... 
> Preparing to unpack .../openshot-qt_2.4.2+dfsg2+941+201807071002~ubuntu18.04.1_all.deb ...
> Unpacking openshot-qt (2.4.2+dfsg2+941+201807071002~ubuntu18.04.1) over (2.4.2+dfsg2+940+201807041607~ubuntu18.04.1) ...
> Processing triggers for mime-support (3.60ubuntu1) ...
> Processing triggers for desktop-file-utils (0.23-1ubuntu3) ...
> Setting up openshot-qt (2.4.2+dfsg2+941+201807071002~ubuntu18.04.1) ...
> Processing triggers for shared-mime-info (1.9-2) ...
> Processing triggers for gnome-menus (3.13.3-11ubuntu1) ...
> 
> user-name@P50:~$ openshot-qt 
> Loaded modules from installed directory: /usr/lib/python3/dist-packages/openshot_qt
>       launch:INFO ------------------------------------------------
>       launch:INFO    OpenShot (version 2.4.2)
>       launch:INFO ------------------------------------------------
> This application failed to start because it could not find or load the Qt platform plugin "xcb"
> in "".
> 
> Available platform plugins are: dxcb, eglfs, linuxfb, minimal, minimalegl, offscreen, vnc, xcb.
> 
> Reinstalling the application may fix this problem.
> Aborted (core dumped)

Based on advice [elsewhere](https://stackoverflow.com/questions/17106315/failed-to-load-platform-plugin-xcb-while-launching-qt5-app-on-linux-without/25437758), I tried adding and sourcing from .bashrc:

> export QT_QPA_PLATFORM_PLUGIN_PATH=/usr/lib/x86_64-linux-gnu/qt5/plugins/platforms

After trying to run openshot-qt again, I got:

> Loaded modules from installed directory: /usr/lib/python3/dist-packages/openshot_qt
>       launch:INFO ------------------------------------------------
>       launch:INFO    OpenShot (version 2.4.2)
>       launch:INFO ------------------------------------------------
> This application failed to start because it could not find or load the Qt platform plugin "xcb"
> in "/usr/lib/x86_64-linux-gnu/qt5/plugins/platforms".
> 
> Available platform plugins are: dxcb (from /usr/lib/x86_64-linux-gnu/qt5/plugins/platforms), eglfs (from /usr/lib/x86_64-linux-gnu/qt5/plugins/platforms), linuxfb (from /usr/lib/x86_64-linux-gnu/qt5/plugins/platforms), minimal (from /usr/lib/x86_64-linux-gnu/qt5/plugins/platforms), minimalegl (from /usr/lib/x86_64-linux-gnu/qt5/plugins/platforms), offscreen (from /usr/lib/x86_64-linux-gnu/qt5/plugins/platforms), vnc (from /usr/lib/x86_64-linux-gnu/qt5/plugins/platforms), xcb (from /usr/lib/x86_64-linux-gnu/qt5/plugins/platforms), dxcb, eglfs, linuxfb, minimal, minimalegl, offscreen, vnc.
> 
> Reinstalling the application may fix this problem.
> Aborted (core dumped)

This is marginally less cryptic, insofar as it is evidently the QT_QPA_PLATFORM_PLUGIN_PATH environment variable that was being referred to by the empty quotes in the original error ('...the Qt platform plugin "xcb" in "".'

This is as much time as I have for problem solving just now, as I'm trying to create the memorial video compilation for my father-in-law's funeral, and I need to find a different solution. I'll follow-up on any further suggestions when I can afterward. Thanks!!

**Comment by peanutbutterandcrackers** [comment]
Sorry about your loss...

You could perhaps try out [Pitivi](http://www.pitivi.org/), [ShotCut](https://www.shotcut.org/), or [Kdenlive](https://www.kdenlive.org/) in the time being. Or perhaps even try the AppImages. That should probably work. But [this workaround](https://github.com/OpenShot/openshot-qt/issues/1757) might be necessary, too.

We'll try to see if anything can be figured out, in the meantime.

**Comment by ferdnyc** [comment]
@rm-minus-r-star The strikethroughs are because `~` is recognized markup in Markdown, and markup is still processed in quoted text. You should use fenced code blocks (three ` ``` ` on a separate line) at the start and end of your pasted logs, to keep it from reading what's between as Markdown.

If your distro's Qt installation doesn't supply `libqxcb.so` (and you can't install it from some additional package), you should be able to run the AppImage builds which are packaged with `libqcb.so`. But I'd try to first find an installable package that provides `/usr/lib/x86_64-linux-gnu/qt5/plugins/platforms/libqxcb.so` (and the PPA should be updated to require that package). It may not match the glob `qt5*` unfortunately.

**Comment by ferdnyc** [comment]
And as @peanutbutterandcrackers said, my condolences re: your father in law.

**Comment by ferdnyc** [comment]
Nope! Sorry, it's in `libqt5gui5`. ([The package listing at Ubuntu](https://packages.ubuntu.com/xenial/amd64/libqt5gui5/filelist))

**Comment by ferdnyc** [comment]
Who's maintaining the Ubuntu PPA, is it anyone we're in direct communication with? If installing OpenShot from the PPA didn't automatically pull in `libqt5gui5` with it, then it really needs to be updated to declare that (apparently required) dependency. I'd like to prevent this same problem coming up for other Ubuntu users, if we can.

**Comment by peanutbutterandcrackers** [comment]
Posted on slack!

**Comment by rm-minus-r-star** [comment]
Many thanks @peanutbutterandcrackers and @ferdnyc -- I actually settled on scripting the movie build with moviepy. Took me longer than it should have to get the Ken Burns thing going, but it looks pretty good in the end.

I had actually also tried the AppImagebut I couldn't get that to run on my (preferred) Ubuntu machine either. I _did_ get it running OK on a Windows laptop, but Windows has it's own special problems and I only had time for one crash before deciding it had to be Linux.

Many thanks again!

**Comment by ferdnyc** [comment]
@rm-minus-r-star Glad you got it worked out, that's the important thing. Yeah, the AppImage issue is a known (completely different) problem, on Ubuntu, and that's in the process of hopefully being addressed. It's unfortunate that the PPA has its own issues at the same time, though. Hopefully we'll be able to get at least one of those in good shape soon.

**Comment by basavaraj1995** [comment]
`sudo rviz'

**Comment by ferdnyc** [comment]
@basavaraj1995  ...?

**Comment by basavaraj1995** [comment]
Try fixing  broken packages.

**Comment by ferdnyc** [comment]
Ah, OK. I'm not a Ubuntu user, so I don't know anything about `rviz`. I'm guessing it's an `apt` tool of some sort?

**Comment by WuYingzheng** [comment]
i suffer from this error for a long time...
`This application failed to start because it could not find or load the Qt platform plugin "xcb"
in "".

Available platform plugins are: dxcb, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx`
following works for me.
`sudo apt --reinstall install libqt5gui5`

**Comment by WuYingzheng** [comment]
i suffer from this error for a long time...
This application failed to start because it could not find or load the Qt platform plugin "xcb"
in "".

Available platform plugins are: dxcb, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx
following works for me.sudo apt --reinstall install libqt5gui5`

**Comment by WuYingzheng** [comment]
i suffer from this error for a long time...
This application failed to start because it could not find or load the Qt platform plugin "xcb"
in "".

Available platform plugins are: dxcb, wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx

following works for me.
`sudo apt --reinstall install libqt5gui5`

**Comment by shyamsreenivasan** [comment]
Yes, issuing the below command fixes the problem on Ubuntu
sudo apt --reinstall install libqt5gui5

**Comment by ferdnyc** [comment]
Ooh, I'm glad you revived this one @shyamsreenivasan. I'm hoping to get the PPA definition updated at some point, and adding `libqt5gui5` as a dependency totally slipped my mind.

**Comment by dospunk** [comment]
I tried running `sudo apt --reinstall install libqt5gui5` and it did not fix the problem for me, I still get the ` Could not find the Qt platform plugin "xcb" in ""` error

**Comment by Colorjet3** [comment]
Please give v3.0.0 a try.  If the problem persists then reopen this ticket.
